source/Calamari/ArgoCD/Conventions/UpdateArgoCDAppImagesInstallConvention.cs

```diff
@@ -185,6 +185,7 @@ public void Install(RunningDeployment deployment)
                 log.InfoFormat(message, linkifiedAppName);
             }
 
+            var gatewayIds = argoProperties.Applications.Select(a => a.GatewayId).ToHashSet();
```

Comment:
Minor side change, moving the loading of gateways to where its used rather than building a hash as a side effect of running the actual operation.

If there ae so many applications that this operation is not performant, there are probably going to be a few other perf problems with this step.